x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LogicalPlanBuilder.java

```diff
@@ -274,7 +275,8 @@ public LogicalPlan visitFromCommand(EsqlBaseParser.FromCommandContext ctx) {
             for (var c : metadataOptionContext.UNQUOTED_SOURCE()) {
                 String id = c.getText();
                 Source src = source(c);
-                if (MetadataAttribute.isSupported(id) == false) {
+                if (MetadataAttribute.isSupported(id) == false
+                    || (EsqlCapabilities.Cap.METADATA_SCORE.isEnabled() == false && MetadataAttribute.SCORE.equals(id))) {
```

Comment:
Might be good to leave a comment that the check can be dropped once `METADATA_SCORE` will no longer be snapshot-only.